vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SearchManager.java

```diff
@@ -0,0 +1,267 @@
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISearchManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+import edu.asu.diging.vspace.core.services.impl.model.StaffSearchModuleResults;
+import edu.asu.diging.vspace.core.services.impl.model.StaffSearchSlideResults;
+import edu.asu.diging.vspace.core.services.impl.model.StaffSearchSlideTextBlockResults;
+import edu.asu.diging.vspace.core.services.impl.model.StaffSearchSpaceResults;
+
+
+@Service
+public abstract class SearchManager implements ISearchManager {
+    
+    @Value("${page_size}")
+    private int pageSize;
+    
+    @Autowired
+    private ISequenceManager sequenceManager;
+    
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private ISlideManager slideManager;
+
+    @Autowired
+    private TextContentBlockRepository textContentBlockRepo;
+    
+    /**
+     * Method to return the requested spaces whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return set of spaces whose name or description contains the search string in
+     *         the requested page.
+     */
+    @Override
+    public Page<ISpace> searchInSpaces(String searchTerm, int page) {
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForSpace = PageRequest.of(page - 1, pageSize);       
+        Page<ISpace> spacePage =  spaceSearch(requestedPageForSpace, searchTerm);
+        
+        int totalSpacePage = spacePage.getTotalPages();
+        /* if page>total pages,last page is returned */
+        if (page > totalSpacePage) {
+            totalSpacePage = totalSpacePage == 0 ? 1 : totalSpacePage;
+            requestedPageForSpace = PageRequest.of(totalSpacePage - 1, pageSize);   
+            spacePage = spaceSearch(requestedPageForSpace,  searchTerm);
+        }
+        return spacePage;
+    }
+    
+    /**
+     * Method to return the requested modules whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return set of modules whose name or description contains the search string
+     *         in the requested page.
+     */
+    @Override
+    public Page<IModule> searchInModules(String searchTerm, int page) {
+        /* if page<1, 1st page is returned */
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForModule = PageRequest.of(page - 1, pageSize);
+        Page<IModule> modulePage = moduleManager.findByNameOrDescription(requestedPageForModule, searchTerm);
```

Comment:
Yes. But if a module is not linked to any space, it doesn't show up in the public search results. Because this method returns an empty list. edu.asu.diging.vspace.core.services.IPublicSearchManager.updateModuleListWithSpaceInfo(Page<IModule>)

Reply:
how is that possible? `findByNameOrDescription` in the module manager has this code:
`return moduleRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage,searchText,searchText);`
that doesn't check if a module is reachable. Or am I missing something?

Reply:
Once findByNameOrDescription returns the modules, only for public search, it goes into updateModuleListWithSpaceInfo() which tries to look for linked spaces, since it doesn't find any spaces, it returns an empty list.

Reply:
but then the pagination is off isn't it?

Reply:
@jdamerow question, in public search , we are only listing modules which are linked to a space. this should be Published space only right? same question for slides and slide texts.